notation.go

```diff
@@ -40,6 +41,9 @@ type SignOptions struct {
 	// PluginConfig sets or overrides the plugin configuration.
 	PluginConfig map[string]string
 
+	// UserMetadata contains key-value pairs that are added to the signature payload
+	UserMetadata map[string]string
```

Comment:
[ocispec.Descirptor](https://pkg.go.dev/github.com/opencontainers/image-spec/specs-go/v1#Descriptor) already has an field of `Annotations`. Should we reuse it?

Reply:
in both cases, the descriptor is fetched in notation.Sign/notation.Verify.
- i think adding annotations to the descriptor before calling signer.Sign makes sense
- for verify, the artifact descriptor passed in is being validated, so re-using the annotations there to pass the user-metadata that we're looking for doesn't make sense

Reply:
Since `notation-go` is a library, developers can invoke `Signer.Sign` and `Verifier.Verify` directly.

As you have mentioned, `notation.Sign` and `notation.Verify` does not have an entrypoint for user metadata. Hence, we do need an option for the user metadata. For `Verify`, we can add it in the `RemoteVerifyOptions`. For `Sign`, we may introduce a new struct `RemoteSignOptions` just like `RemoteVerifyOptions`.

Reply:
@byronchien, I might have mislead you.
I agree with @shizhMSFT, we are already passing a descriptor for the `Sign` method so we don't need a new metadata variable in `notation.SignOption`. Instead, what can pass a descriptor with metadata already filled in as annotations.

Reply:
ok, i think i understand now, will add a RemoteSignOptions, remove usermetadata from SignOptions, and add the annotations to the descriptor before Signer.Sign